### PR TITLE
frontend: use Context for Darkmode

### PR DESCRIPTION
**Pull request by shonsirsha**
Using context for Darkmode allows us to get and set the darkmode state easier across component (globally). We used our own `useDarkmode` hook to wrap the dark mode context.

Keeping `getDarkmode()` as it's still required by class components.

### CODE REVIEW

File: backend/chart.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func (backend *Backend) ChartData() (*Chart, error) {
 		chartDataMissing = true
 		backend.log.Info("ChartDataMissing, until is zero")
 	}
-	isUpToDate := time.Since(until) < 2*time.Hour
+	isUpToDate := time.Since(until) < 24*time.Hour
```
Comment (thisconnect) [inline backend/chart.go:127]: this is unrelated (I think) remove?

Comment (shonsirsha) [reply backend/chart.go:127]: Ooh lol. I was testing. Will remove.

**Review by thisconnect — CHANGES_REQUESTED** [review]
nice 👍

File: frontends/web/src/contexts/DarkmodeProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }, [osPrefersDarkmode]);
+
+  useEffect(() => {
+    if (isDarkMode) {
+      document.body.classList.add('dark-mode');
+      document.body.classList.remove('light-mode');
+    } else {
+      document.body.classList.remove('dark-mode');
+      document.body.classList.add('light-mode');
+    }
```
[the remark this answers is not in the document]
Comment (shonsirsha) [reply frontends/web/src/contexts/DarkmodeProvider.tsx:36]: Ah good catch. We don't need the other file's classList changes. We just need the setting of `darkmode`. E.g

```js
let darkmode: boolean | undefined;
const { isDarkMode } = useDarkmode();
darkmode = isDarkMode;
```

will make the change.

**Comment by shonsirsha** [comment]
Thanks for the review @thisconnect! Changes made, PTAL  🙏

File: frontends/web/src/app.tsx [hunk 3]
```diff
@@ -184,28 +185,30 @@ class App extends Component<Props, State> {
...
+              />
+              <RouterWatcher onChange={this.handleRoute} />
+            </div>
+            <Alert />
+            <Confirm />
           </div>
-          <Alert />
-          <Confirm />
-        </div>
+        </DarkModeProvider>
```
Comment (thisconnect) [inline frontends/web/src/app.tsx:211]: one more context-noob question: if we wanted to move sidebar state and toggling + guide state and toggling to context, would that be another provider here? i.e. `<AppProvider>` or similar?

**Review by thisconnect — COMMENTED** [review]
added one more comment, also added @Beerosagos and @benma to take a look since this is the first react context in the BitBoxApp

**Comment by shonsirsha** [comment]
thanks for the review! @thisconnect

**Comment by thisconnect** [comment]
One more to convert (if it is merged first).

**Review by benma — COMMENTED** [review]
Is this strictly a refactoring or does it change something functionally (fix a bug or sth)?

**Comment by shonsirsha** [comment]
> Is this strictly a refactoring or does it change something functionally (fix a bug or sth)?

This also fixes a bug where the theme weirdly toggles on its own when language gets changed.

**Comment by shonsirsha** [comment]
> One more to convert in https://github.com/digitalbitbox/bitbox-wallet-app/pull/2049 (if it is merged first).

Nice. I think it makes sense to merge that one first, then change it to be using hook in this PR, then merge this one :) wdyt?

**Comment by thisconnect** [comment]
ack. merged, please rebase and change globe icon

> Nice. I think it makes sense to merge that one first, then change it to be using hook in this PR, then merge this one :) wdyt?

**Comment by shonsirsha** [comment]
> ack. merged, please rebase and change globe icon

Actually I dont think we need to make any more changes because of your PR. Since your PR is using `useDarkmode`, and `useDarkmode` is already using context here :) 

I'll still make some changes to fix the conflict due to  #2035 .

File: frontends/web/src/contexts/DarkmodeProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  const setAppTheme = useCallback(() => {
+    setDarkTheme(isDarkMode);
+    if (isDarkMode) {
+      document.body.classList.add('dark-mode');
+      document.body.classList.remove('light-mode');
+    } else {
+      document.body.classList.remove('dark-mode');
+      document.body.classList.add('light-mode');
+    }
+  }, [isDarkMode]);
```
Comment (shonsirsha) [inline frontends/web/src/contexts/DarkmodeProvider.tsx:40]: Here we used the new `setDarkTheme` created in #2035 . CCing @Beerosagos and @thisconnect in case wanting to take a look once more before I merge.

Comment (shonsirsha) [reply frontends/web/src/contexts/DarkmodeProvider.tsx:40]: also wrapped it in a function `setAppTheme` just so it's neater:)

**Comment by thisconnect** [comment]
@shonsirsha please rebase, then CI should run properly.
